bin/runtime-common/src/messages_xcm_extension.rs

```diff
@@ -158,11 +126,7 @@ impl<
 				XcmBlobMessageDispatchResult::NotDispatched(e)
 			},
 		};
-		MessageDispatchResult {
-			// TODO:check-parameter - setup uspent_weight?  https://github.com/paritytech/polkadot/issues/6629
```

Comment:
> Weights::message_dispatch_weight(payload_size) means now worst scenario, right?

Yes, it should be the worst scenarion. And in our case (we always push message to the XCMP queue), it is the only scenario. If we'll be adding more options for message routing, then it may change. But for now it is fine I believe

Reply:
well, if we bridge message for relaychain as target then it would be DmpQueue, XcmpQueue is for parachain as target,
but I guess it is similar and XcmpQueue is worst case

ayway, do we still need `unspent_weight`?

Reply:
Yeah - that's a good point. If we'll be supporting to-relay routers or e.g. to-ethereum routers, then we'll probably want them to return actual dispatch weight (that's an ideal option). Then we may return actual `unspent_weight` and make things cheaper (and fit more messages into bridge hub block).

An alternate option may just use some runtime-level (or specific router-level) benchmarks to compute this kinda actual weight. So we'll be returning the worst possible value from `dispatch_weight` function and then from actual dispatch we will be looking at the XCM destination and return proper `unspent_weight`.

But for now, at least until we start work on dynamic bridging stuff, I think it is fine to have this ^^^ implementation